packages/rn-tester/js/examples/TextInput/TextInputSharedExamples.js

```diff
@@ -122,18 +120,40 @@ class RewriteExampleInvalidCharacters extends React.Component<
 > {
   constructor(props) {
     super(props);
-    this.state = {text: ''};
+    this.state = {text: '', selection: {start: -1, end: -1}};
   }
+
+  onSelectionChangeHandler = ({nativeEvent: {selection}}) => {
+    const {start, end} = selection;
+    const {text} = this.state;
+    const maxEnd = text.length;
+    const validEnd = end > maxEnd ? maxEnd : end;
+    const validStart = start > maxEnd ? maxEnd : start;
+    const newSelection = {start: validStart, end: validEnd};
+    this.setState({selection: newSelection});
+  };
```

Comment:
as displayed in the below GIF, [this solution][1] is not acceptable as it would create a different behaviour between iOS and Android. Android would need to set the selection position (the cursor) in this type of use case scenario, while the same solution on iOS would present different behaviour adding further complexity to the framework. I believe that I will need to add logic to TextInput useEffect/useLayout callbacks to handle this different scenarios, without regressions or difference in behaviour between iOS and Android.

The below gifs displays the issue experienced on iOS. Thanks a lot!

<img src="https://user-images.githubusercontent.com/24992535/94347595-e890c300-0035-11eb-843f-593f9091ae71.gif" width="300" />

[1]: https://github.com/facebook/react-native/pull/29070/files#diff-51e729971c02a077b6aa6004042b7b61R126-R134